### ISSUE TITLE
Azure ML identity issues with and without managed identity

### ISSUE THREAD

**Issue by bloewes** [Azure/azure-sdk-for-python#31384]
- **Package Name**: azure-identity
- **Package Version**: 1.13.0
- **Operating System**: 20.04.1-Ubuntu
- **Python Version**: 3.8.5

**Describe the bug**
We are trying to use DefaultAzureCredential on an Azure ML compute 
1. without any assigned identity
2. with user assigned managed identity

**Additional context**
For both the following code is used 
```
from azure.identity import DefaultAzureCredential
myid = DefaultAzureCredential()
myid.get_token()
```
1. Without any assigned identity I get the following output:
```
DefaultAzureCredential failed to retrieve a token from the included credentials.
Attempted credentials:
        EnvironmentCredential: EnvironmentCredential authentication unavailable. Environment variables are not fully configured.
Visit https://aka.ms/azsdk/python/identity/environmentcredential/troubleshoot to troubleshoot this issue.
        ManagedIdentityCredential: "get_token" requires at least one scope
To mitigate this issue, please refer to the troubleshooting guidelines here at https://aka.ms/azsdk/python/identity/defaultazurecredential/troubleshoot.
Traceback (most recent call last):
  File "identity.py", line 23, in <module>
    myid.get_token()
  File "/anaconda/envs/azureml_py38/lib/python3.8/site-packages/azure/identity/_credentials/default.py", line 214, in get_token
    return super(DefaultAzureCredential, self).get_token(*scopes, **kwargs)
  File "/anaconda/envs/azureml_py38/lib/python3.8/site-packages/azure/identity/_credentials/chained.py", line 111, in get_token
    raise ClientAuthenticationError(message=message)
azure.core.exceptions.ClientAuthenticationError: DefaultAzureCredential failed to retrieve a token from the included credentials.
Attempted credentials:
        EnvironmentCredential: EnvironmentCredential authentication unavailable. Environment variables are not fully configured.
Visit https://aka.ms/azsdk/python/identity/environmentcredential/troubleshoot to troubleshoot this issue.
        ManagedIdentityCredential: "get_token" requires at least one scope
To mitigate this issue, please refer to the troubleshooting guidelines here at https://aka.ms/azsdk/python/identity/defaultazurecredential/troubleshoot.
```
I saw in https://github.com/Azure/azure-sdk-for-python/blob/7f46b4eeb69a67d74a97848e265750b85afde5a2/sdk/identity/azure-identity/azure/identity/_credentials/managed_identity.py#L65-L70 that if both MSI_ENDPOINT and MSI_SECRET are set as env variables, which they are if I check from the terminal, Azure ML MSI is used. Now my question: Should they be set in this case, or, if they are always set by Azure ML, is the code potentially not covering cases where no identity is assigned. 
Bottom line: I want to use Azure CLI identification here, but I don't want to hardcode, as they code base is also used for machines with assigned identity.

2. Even with assigned identity, the call fails. It only works, if I manually add the managed_identity_client_id of the assigned identity in the `DefaultAzureCredential(managed_identity_client_id=xxx)`

**Comment by kashifkhan** [comment]
Thanks for the feedback @bloewes . We will investigate and get back to you asap.

**Comment by xiangyan99** [comment]
Thanks for reaching out.

Let's take your questions one by one. :)
1.  The environment variables. They should be set by Azure. If you create a managed VM in Azure ML and assigned a managed identity, those environment variables would be set automatically. They should not be set manually.
2. In your case, you used a user-assigned managed identity and you wanted to specify the id of it. Then you can use managed_identity_client_id to achieve it.
3. Failure of myid.get_token(). This is by design. The API signature of get_token() is get_token(*scope), you need to specify which resource you want to access. If you use our SDK libraries, e.g. keyvault, storage, etc., we specify the scope information for you then you don't need to take care of it. If you want to call it by yourself, you need to get the scope information and pass it into get_token() method.
4. You can use DefaultAzureCredential. It is a chain of credentials. (https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/README.md#defaultazurecredential). If other credentials do not work, it uses cli credentials in the chain.

Please feel free to let us know if you have more questions.
